sdk/v1/src/main/java/com/google/cloud/trace/v1/sink/ScheduledBufferingTraceSink.java

```diff
@@ -68,17 +68,19 @@ public ScheduledBufferingTraceSink(TraceSink traceSink, Sizer<Trace> traceSizer,
   }
 
   @Override
-  public void receive(Trace trace) {
+  public void receive(Traces traces) {
     synchronized(monitor) {
-      traceBuffer.put(trace);
-      size += traceSizer.size(trace);
-      if (size >= bufferSize) {
-        if (autoFlusher == null) {
-          autoFlusher = scheduler.submit(flushable());
-        }
-      } else {
-        if ((flusher == null) && (autoFlusher == null)) {
-          flusher = scheduler.schedule(flushable(), scheduledDelay, TimeUnit.SECONDS);
+      for (Trace trace : traces.getTracesList()) {
```

Comment:
I know I didn't do a very good job of this, but it would be nice to move the TraceBuffer translation logic outside of the synchronized block.